Firestore/core/src/model/value_util.cc

```diff
@@ -448,19 +449,17 @@ std::string CanonifyObject(const google_firestore_v1_Value& value) {
   pb_size_t fields_count = value.map_value.fields_count;
   const auto& fields = value.map_value.fields;
 
-  // Porting Note: MapValues in iOS are always kept in sorted order. We
-  // therefore do no need to sort them before generating the canonical ID.
-  std::string result = "{";
+  std::vector<std::string> fields_string;
   for (pb_size_t i = 0; i < fields_count; ++i) {
-    absl::StrAppend(&result, nanopb::MakeStringView(fields[i].key), ":",
+    std::string field;
+    absl::StrAppend(&field, nanopb::MakeStringView(fields[i].key), ":",
                     CanonicalId(fields[i].value));
-    if (i != fields_count - 1) {
-      absl::StrAppend(&result, ",");
-    }
+    fields_string.push_back(std::move(field));
   }
-  result += "}";
+  std::sort(fields_string.begin(), fields_string.end());
```

Comment:
Why did this have to change?

The sort order is also slightly off in some cases as you are sorting by "field:value" rather than just field, which will yield a different sort order for fields that contain special characters - and that could be considered a breaking change as these queries will not be able to served from cache right after the SDK upgrade (probably not a problem in practice).